lib/Prefs/Special/SmimePrivateKey.php

```diff
@@ -35,45 +35,83 @@ public function display(Horde_Core_Prefs_Ui $ui)
     {
         global $browser, $injector, $page_output, $prefs;
 
+        /* Adding js to page output */
         $page_output->addScriptPackage('IMP_Script_Package_Imp');
 
+        /* Adding css to page output */
         $p_css = new Horde_Themes_Element('prefs.css');
         $page_output->addStylesheet($p_css->fs, $p_css->uri);
 
+        /* an instance of IMP_smime to be able to list private keys from the DB */
+        $smime = $injector->getInstance('IMP_Smime');
+        try {
+            $extra_private_keys = $smime->listPrivateKeys();
+        } catch (Horde_Exception $e) {
+            $extra_private_keys = [];
+        }
+
+        /* Loading View Template and Help Template */
         $view = new Horde_View(array(
             'templatePath' => IMP_TEMPLATES . '/prefs'
         ));
         $view->addHelper('Horde_Core_View_Helper_Help');
 
+        /* Loading Connection Status to View */
         if (!Horde::isConnectionSecure()) {
             $view->notsecure = true;
             return $view->render('smimeprivatekey');
         }
 
-        $smime_url = IMP_Basic_Smime::url();
 
+        /* Loading Keys that are set as Personal Certificate
+         (the certificates that are actually used) */
         $view->has_key = $prefs->getValue('smime_public_key') &&
             $prefs->getValue('smime_private_key');
         $view->has_sign_key = $prefs->getValue('smime_public_sign_key') &&
             $prefs->getValue('smime_private_sign_key');
 
+        /* Loading Smime bas url in order to set links to it */
+        $smime_url = IMP_Basic_Smime::url();
+
+        /* Addding to view: Browser Importoptions for uploading Certificates */
         if ($browser->allowFileUploads()) {
             $view->import = true;
             $page_output->addInlineScript(array(
                 '$("import_smime_personal").observe("click", function(e) { ' . Horde::popupJs($smime_url, array('params' => array('actionID' => 'import_personal_certs', 'reload' => base64_encode($ui->selfUrl()->setRaw(true))), 'height' => 450, 'width' => 750, 'urlencode' => true)) . '; e.stop(); })'
             ), true);
         }
+
+        /* Loading private keys from the Database that are not used as Personal Certificates */
+        if (!empty($extra_private_keys)) {
+            // adding base url links to each private key: so one can view the keys
+            $pk_list = [];
+            foreach ($extra_private_keys as $val) {
+                $link = $smime_url->copy()->add(['actionID' => 'view_extra_private_keys', 'pkID' => $val['private_key_id']]);
+                $tile = "View Extra Private Keys";
+                $pk_list[] = Horde::link($link, $title, null, 'view_key'); // all uneven keys hold links
+                $pk_list[] = $val['private_key_id'];    // all even links hold privatekey ids
+            }
+
+            // Adding extra private keys to view
+            $view->{'viewprivateextras'} = $pk_list;
```

Comment:
it add an attribute to the $view object called 'viewprivateextras' which contains an array as defined in the lines directly above.
 
I reused this syntax because thats the way it has been done in imp